query-coordinator/src/main/scala/com/socrata/querycoordinator/resources/QueryResource.scala

```diff
@@ -336,8 +336,14 @@ class QueryResource(secondary: Secondary,
               case PipeQuery(l, r) =>
                 val (nl, rollupLeft) = possiblyRewriteOneAnalysisInQuery(schema, l, Some(ruMap))
                 val (nr, rollupJoin) = possiblyRewriteJoin(r)
-                val rewritten = (PipeQuery(nl, nr), rollupLeft ++ rollupJoin)
-                if (ruMapOpt.isEmpty && ruMap.nonEmpty && rollupLeft.isEmpty) {
+                val (nr2, rollupRight) = possiblyRewriteOneAnalysisInQuery(schema, nr, Some(ruMap))
+                val rewritten = ((rollupLeft,rollupJoin,rollupRight)match{
+                  //If Join or Right are rewritten, then its officially the "right" side, only right matters
+                  case (Nil,Nil,Seq(_)) | (Nil,Seq(_),Nil) | (Nil,Seq(_),Seq(_))=>nr2
```

Comment:
If we rewrote any combination of the "join" or "right"(which is itself the "join"), then we only care about the right side, the left has no data for us that matters.

Otherwise, return both left+right as PipeQuery